awscli/customizations/s3/s3handler.py

```diff
@@ -513,3 +521,187 @@ def _process_transfer(self, future):
             # TODO: Update when S3Handler is refactored
             uni_print("Transfer failed: %s \n" % str(e))
             return CommandResult(1, 0)
+
+
+class BaseTransferRequestSubmitter(object):
+    REQUEST_MAPPER_METHOD = None
+    RESULT_SUBSCRIBER_CLASS = None
+
+    def __init__(self, transfer_manager, result_queue, cli_params):
+        """Submits transfer requests to the TransferManager
+
+        Given a FileInfo object and provided CLI parameters, it will add the
+        necessary extra arguments and subscribers in making a call to the
+        TransferManager.
+
+        :param transfer_manager: The underlying transfer manager
+        :param result queue: The result queue to use
+        :param cli_params: The associated CLI parameters passed in to the
```

Comment:
Would be helpful to document the type.  Is this the `Namespace` obj from argparse?